corehq/apps/app_manager/decorators.py

```diff
@@ -66,13 +68,22 @@ def _safe_cached_download(request, *args, **kwargs):
         # make endpoints that call the user fail hard
         from django.contrib.auth.models import AnonymousUser
         request.user = AnonymousUser()
+        username = request.GET.get('username')
         if request.GET.get('username'):
             request.GET = request.GET.copy()
             request.GET.pop('username')
 
         latest_enabled_build = None
-        if latest and request.GET.get('profile') and toggles.RELEASE_BUILDS_PER_PROFILE.enabled(domain):
-            latest_enabled_build = get_latest_enabled_build_for_profile(domain, request.GET.get('profile'))
+        if latest and toggles.MANAGE_RELEASES_PER_LOCATION.enabled(domain):
+            if not username:
+                content_response = dict(error="app.update.not.allowed.user.logged_out",
+                                        default_response="Please log in the app to check for update.")
```

Comment:
translate. Also, I'd change this to "Please log in _to_ the app to check for _an_ update."